src/deadline/job_attachments/asset_sync.py

```diff
@@ -225,7 +225,15 @@ def aggregate_asset_root_manifests(
                     local_root = str(session_dir.joinpath(dir_name))
                     grouped_manifests_by_root[local_root].extend(manifests)
 
-        return grouped_manifests_by_root
+        # Merge the manifests in each root into a single manifest
```

Comment:
Are we doing this refactoring so Worker Agent works on a similar input code path? EG: new code path uses merged manifests and this `aggregate_asset_root_manifests ` function, and the old code path use `sync_input` as a monolithic function ? 

(Just providing context)